clippy_lints/src/panic_unimplemented.rs

```diff
@@ -0,0 +1,88 @@
+use rustc::hir::*;
+use rustc::lint::*;
+use syntax::ast::LitKind;
+use syntax::ptr::P;
+use utils::{is_direct_expn_of, is_expn_of, match_def_path, opt_def_id, paths, resolve_node, span_lint};
+
+/// **What it does:** Checks for missing parameters in `panic!`.
+///
+/// **Why is this bad?** Contrary to the `format!` family of macros, there are
+/// two forms of `panic!`: if there are no parameters given, the first argument
+/// is not a format string and used literally. So while `format!("{}")` will
+/// fail to compile, `panic!("{}")` will not.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// panic!("This `panic!` is probably missing a parameter there: {}");
+/// ```
+declare_clippy_lint! {
+    pub PANIC_PARAMS,
+    style,
+    "missing parameters in `panic!` calls"
+}
+
+/// **What it does:** Checks for usage of `unimplemented!`.
+///
+/// **Why is this bad?** This macro should not be present in production code
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// unimplemented!();
+/// ```
+declare_clippy_lint! {
+    pub UNIMPLEMENTED,
+    style,
```

Comment:
I think this should be a restriction lint, as we don't have a group for "lint this only in release mode".

It's definitely not a `style` lint. I don't know what it is though.

Then again, rustc lints unused functions by default. So we might just want to throw this into the `clippy` group and not anywhere else. Opinions anyone?